### PR TITLE
Add ability to set rsa public key for user resource and add schema resource

### PR DESCRIPTION
**Pull request by wookasz**
**What**
- Add optional rsa_public_key parameter
- Ability to create a schema in database
- Ability to update by changing name or moving schema to another database
- Ability to delete schema

**Why**
We want to be able to do both with terraform

**Testing**
NMTN - tested by applying changes locally

**TF Plan**
```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + snowflake_schema.test
      id:                 <computed>
      database:           "ELT_TEST"
      schema:             "TEST"

  + snowflake_user.test
      id:                 <computed>
      default_role:       "ELT_TEST"
      plaintext_password: <sensitive>
      rsa_public_key:     "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA5iZVDwijNQgtd83cJa5kCL+4pJA/JeRq1ihnCH+JZ1S8IAQGgfwrPtkvycckR3E5ZOhjjYvOTck+6OrYMvQzz7Iiy0n3ZUvR29h42n72tfaVCzd3gu1D7djLVuDBn5c7hjGUUavwYK37P/s1kY+l3llwcZlMIsRhoR/D2dWtGQP48qukvddhxw/08xD26sVSrS2h3azCbFBPhxoZK0vBY2OZHmdOiNoQp8zJ3rfHUzuUrcqsUFgM9KbISEFBATf9YQA1T/3e+2Q1h2SCqlbHt00EvFjyJ8c7lJlsLKEUT4JTLcQN9F3/Zflv5nURBEimbYI6JwOA0yeb/bV2zBVLMQIDAQAB"
      user:               "TESTER"


Plan: 2 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------
```

based on this spec:

```
resource "snowflake_schema" "test" {
  database = "ELT_TEST"
  schema   = "TEST"
}

resource "snowflake_user" "test" {
  user = "TESTER"
  default_role = "ELT_TEST"
  plaintext_password = "changeme12345!"
  rsa_public_key = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA5iZVDwijNQgtd83cJa5kCL+4pJA/JeRq1ihnCH+JZ1S8IAQGgfwrPtkvycckR3E5ZOhjjYvOTck+6OrYMvQzz7Iiy0n3ZUvR29h42n72tfaVCzd3gu1D7djLVuDBn5c7hjGUUavwYK37P/s1kY+l3llwcZlMIsRhoR/D2dWtGQP48qukvddhxw/08xD26sVSrS2h3azCbFBPhxoZK0vBY2OZHmdOiNoQp8zJ3rfHUzuUrcqsUFgM9KbISEFBATf9YQA1T/3e+2Q1h2SCqlbHt00EvFjyJ8c7lJlsLKEUT4JTLcQN9F3/Zflv5nURBEimbYI6JwOA0yeb/bV2zBVLMQIDAQAB"
}
```

achieves these results:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,35 +67,34 @@ $ export SF_ACCOUNT
...
+| `statement_queued_timeout_in_seconds` | Time, in seconds, an SQL statement can be queued before being cancelled | Integer | FALSE |
+| `statement_timeout_in_seconds` | Time, in seconds, after which an SQL statement will be terminated | Integer | FALSE |
 | `warehouse_size` | Size of the warehouse | String | FALSE |
-| `max_cluster_count` | Min number of warehouses | String | FALSE |
-| `min_cluster_count` | Max number of warehouses | String | FALSE |
+| `max_cluster_count` | Min number of warehouses | Integer | FALSE |
+| `min_cluster_count` | Max number of warehouses | Integer | FALSE |
 | `auto_resume` | Should warehouse should auto resume | Boolean | FALSE |
-| `auto_suspend` | Should warehouse should auto suspend | Integer | 600 |
+| `auto_suspend` | Should warehouse should auto suspend | Integer | FALSE |
```
Comment (malexovi) [inline README.md]: Boolean?

Comment (wookasz) [reply README.md]: yes!

File: snowflake/resource_schema.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+			"database": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the database in which to place schema",
+			},
+
+			"schema": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the schema",
```
Comment (malexovi) [inline snowflake/resource_schema.go]: Nit: inconsistent phrasing

File: snowflake/resource_schema.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	for rows.Next() {
+		var dbCreatedAt string
+		var dbName string
+		var dbKind sql.NullString
+		var dbDBName string
+		var dbSchemaName sql.NullString
+		if err := rows.Scan(&dbCreatedAt, &dbName, &dbKind, &dbDBName, &dbSchemaName); err != nil {
+			return err
+		}
+		if database == dbDBName && schema == dbName {
```
Comment (malexovi) [inline snowflake/resource_schema.go:81]: Suggest adding a comment for why were doing this

File: snowflake/resource_schema.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+			d.Set("schema", dbName)
+			d.Set("database", dbDBName)
+			return nil
+		}
+	}
+
+	return fmt.Errorf("The schema %s.%s does not exist.", database, schema)
```
Comment (malexovi) [inline snowflake/resource_schema.go]: Styleguide says don't capitalize errors

File: snowflake/resource_schema.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	db := meta.(*providerConfiguration).DB
+
+	database, schema := paramsFromSchemaID(d.Id())
+
+	stmtSQL := fmt.Sprintf("DROP SCHEMA \"%s\".\"%s\"", database, schema)
+
+	log.Println("Executing statement:", stmtSQL)
+
+	_, err := db.Exec(stmtSQL)
+	if err == nil {
```
Comment (malexovi) [inline snowflake/resource_schema.go]: Nit: unconventional error handling

**Review by malexovi — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (wookasz) [reply README.md]: Actually, no, integer, but the description here is way wrong

**Pull request merged**